packages/devextreme/js/__internal/ui/toolbar/internal/toolbar.menu.ts

```diff
@@ -282,8 +282,28 @@ export default class DropDownMenu extends Widget<DropDownMenuProperties> {
       dragEnabled: false,
       showTitle: false,
       fullScreen: false,
+      ignoreChildEvents: false,
       _fixWrapperPosition: true,
     });
+    this._popup.registerKeyHandler('space', (
+      e: DxEvent<KeyboardEvent>,
+    ) => {
+      this._popupKeyHandler(e);
+    });
+    this._popup.registerKeyHandler('enter', (
+      e: DxEvent<KeyboardEvent>,
+    ) => {
+      this._popupKeyHandler(e);
+    });
+    this._popup.registerKeyHandler('escape', (
+      e: DxEvent<KeyboardEvent>,
+    ) => {
+      // @ts-expect-error
+      if (this._popup?.$overlayContent().is($(e.target))) {
+        this.option('opened', false);
+      }
```

Comment:
Hm, why do we need this? Looks like a default popup handler 🤔

Reply:
Yes, we need this to minimize changes in behavior after the fix. The task to improve this behavior was moved to UI/UX Toolbar Enhancements, and such a change is better implemented only in the new version

Reply:
Okay I gonna trust you